Dockerfile

```diff
@@ -1,23 +1,14 @@
 # prepare machine
 FROM ubuntu:22.04 as builder
 
-RUN apt update && apt install -y software-properties-common
-
-RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y
-
-RUN apt update && apt install -y \
```

Comment:
Actually sync and install on the same line is a good idea no ? e.g.:
https://stackoverflow.com/a/61439544
Keep it on the same line ?

Reply:
Yep. I always group `apt udpate` and `apt install` in docker files to avoid the problem mentionned in the link you sent.
The thing you saw in the dockerfile that I reverted was to install the latest version of gcc and llvm for testing. Maybe this is something we may reuse in the future (like for testing with different versions of llvm)?